staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -341,6 +352,11 @@ var newETCD3Client = func(c storagebackend.TransportConfig) (*clientv3.Client, e
 		}
 		dialOptions = append(dialOptions, grpc.WithContextDialer(dialer))
 	}
+	if c.EnableGrpcHealthcheck {
+		// ignore config set by etcd manual resolver inside etcd client
+		dialOptions = append(dialOptions, grpc.WithDisableServiceConfig())
+		dialOptions = append(dialOptions, grpc.WithDefaultServiceConfig(grpcServiceConfigWithHealthcheck))
```

Comment:
This is just 1 out of 2 steps to enable gRPC health check on client side and it is not enough solely. 

There is another step which implicitly registers the `clientHealthCheck` and dials to `/grpc.health.v1.Health/Watch` under the hood. 

```
// import grpc/health to enable transparent client side checking 
import _ "google.golang.org/grpc/health"
```

```
func init() {
	internal.HealthCheckFunc = clientHealthCheck
}
```
https://github.com/grpc/grpc-go/blob/master/health/client.go#L51-L53